src/main/java/io/scim2/swagger/client/ScimApiClient.java

```diff
@@ -973,143 +1125,165 @@ public <T> T handleResponse(Response response, Type returnType) throws ScimApiEx
+    public HttpUriRequest buildRequest(String method, List<Pair> queryParams, Object body,
+                                       Map<String, String> headerParams,
+                                       Map<String, Object> formParams, String[] authNames) throws ScimApiException {
 
         updateParamsForAuth(authNames, queryParams, headerParams);
         String path = this.url;
-        final String url = buildUrl(path, queryParams);
-        final Request.Builder reqBuilder = new Request.Builder().url(url);
-        processHeaderParams(headerParams, reqBuilder);
+        final String requestUrl = buildUrl(path, queryParams);
+        final HttpUriRequest request = createHttpRequest(method, requestUrl);
+        processHeaderParams(headerParams, request);
         String contentType = headerParams.get("Content-Type");
         // ensuring a default content type
         if (contentType == null) {
-            contentType = jsonContentType;
+            contentType = JSON_CONTENT_TYPE;
         }
 
-        RequestBody reqBody;
-        if (!HttpMethod.permitsRequestBody(method)) {
-            reqBody = null;
-        } else if (urlEncodedContentType.equals(contentType)) {
-            reqBody = buildRequestBodyFormEncoding(formParams);
-        } else if (multiparFormDataContentType.equals(contentType)) {
-            reqBody = buildRequestBodyMultipart(formParams);
-        } else if (body == null) {
-            if ("DELETE".equals(method)) {
-                // allow calling DELETE without sending a request body
-                reqBody = null;
+        HttpEntity reqBody;
+        if (request instanceof HttpEntityEnclosingRequestBase) {
+            HttpEntityEnclosingRequestBase entityRequest = (HttpEntityEnclosingRequestBase) request;
+            if (URL_ENCODED_CONTENT_TYPE.equals(contentType)) {
+                reqBody = buildRequestBodyFormEncoding(formParams);
+            } else if (MULTIPART_FORM_DATA_CONTENT_TYPE.equals(contentType)) {
+                reqBody = buildRequestBodyMultipart(formParams);
+            } else if (body == null) {
+                // use an empty request body (for POST, PUT, and PATCH)
+                // no request body is allowed for DELETE
+                reqBody = new StringEntity("", ContentType.parse(contentType));
             } else {
-                // use an empty request body (for POST, PUT and PATCH)
-                reqBody = RequestBody.create(MediaType.parse(contentType), "");
+                reqBody = serialize(body, contentType);
             }
-        } else {
-            reqBody = serialize(body, contentType);
+            entityRequest.setEntity(reqBody);
         }
-        Request request = reqBuilder.method(method, reqBody).build();
         return request;
     }
 
+    private HttpUriRequest createHttpRequest(String method, String url) {
+
+        switch (method) {
+            case "GET":
```

Comment:
We could define a enum for this? Or constants. Also check if we have any other library which provide these methods as constants?